en/advanced_features/traffic_avoidance_adsb.md

```diff
@@ -94,7 +94,61 @@ The code can be found in `Navigator::check_traffic` ([/src/modules/navigator/nav
 PX4 will also forward the transponder data to a GCS if this has been configured for the MAVLink instance (this is recommended).
 The last 10 Digits of the GUID is displayed as Drone identification.
 
+## Testing/Simulated ADSB Traffic
+
+You can simulate ADS-B traffic for testing.
+Note that this requires that you [Build PX4](../dev_setup/building_px4.md)
+
+:::note
+Simulated ADS-B traffic can trigger real failsafe actions.
+Use with care in real flight!
+:::
+
+To enable this feature:
+
+1. Uncomment the code in `AdsbConflict::run_fake_traffic()`([AdsbConflict.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/lib/adsb/AdsbConflict.cpp#L342C1-L342C1)).
+1. Rebuild and run PX4.
+1. Execute the [`navigator fake_traffic` command](../modules/modules_controller.md#navigator) in the [QGroundControl MAVLink Shell](https://docs.qgroundcontrol.com/master/en/analyze_view/mavlink_console.html) (or some other [PX4 Console or MAVLink shell](../debug/consoles.md), such as the PX4 simulator terminal).
+
+The code in `run_fake_traffic()` is then executed.
+You should see ADS-B warnings in the Console/MAVLink shell, and QGC should also show an ADS-B traffic popup.
+
+By default `run_fake_traffic()` publishes a number of traffic messages (it calls [`AdsbConflict::fake_traffic()`](#fake-traffic-method) to emit each report).
+These simulate ADS-B traffic where there may be a conflict, where there won't be a conflict, as well as spamming the traffic buffer.
+
+::: details Information about the test methods
+
+The relevent methods are defined in [AdsbConflict.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/lib/adsb/AdsbConflict.cpp#L342C1-L342C1).
+
+#### `run_fake_traffic()` method
+
+The `run_fake_traffic()` method is run when the `navigator fake_traffic` command is called.
+
+The method simulates different scenarios by calling the `fake_traffic()` method with different parameters.
+It creates fake traffic messages with different callsigns, distances, directions, altitude differences, velocities, and emitter types.
+
+The default scenarios include conflicts and non-conflicts, as well as spamming the traffic buffer.
+
+#### `fake_traffic()` method
+
+`AdsbConflict::fake_traffic()` is called by the [`run_fake_traffic()`](#run-fake-traffic-method) to create individual ADS-B transponder reports.
+
+This takes several parameters, which specify the characteristics of the fake traffic: callsign, distance, direction, traffic heading, altitude difference, horizontal velocity, vertical velocity, emitter type, ICAO address, UAV latitude and longitude, and UAV altitude.
+The method modifies the `alt_uav` parameter by reference.
```

Comment:
When you talk about modifying the parameter by reference, what is the point you are trying to make? I.e. why should I care?

Reply:
@farhangnaderi This is still confusing - I think because I don't understand what your algorithm (as defined below) is supposed to achieve. Possibly because the in-source docs were removed :-(

I think: 
- `run_fake_traffic()` passes in the position of the current UAV for which you want to generate fake traffic. That essentially gives a starting point
- `run_traffic()` takes these values right? (see questions)

  ```
	 * @param callsign Callsign of the fake transponder
	 * @param distance Horizontal distance to the fake vehicle from the current vehicle. - WHAT UNITS?
	 * @param direction Direction in NED (?) from this vehicle to the fake in radians
	 * @param traffic_heading Travel direction of the traffic in NED in radians
	 * @param altitude_diff Altitude difference, positive is up of the fake traffic. WHAT UNITS? (so positive number indicates above?)
	 * @param hor_velocity Horizontal velocity of fake traffic, in m/s
	 * @param ver_velocity Vertical velocity of fake traffic, in m/s
	 * @param emitter_type, Type of fake vehicle, as a number
	 * @param icao_address, ?
	 * @param lat_uav, Lat of this vehicle
	 * @param lon_uav, Lon of this vehicle
	 * @param alt_uav, Altof this vehicle (as ref)
  ```
- I think that you pass in the position of the current vehicle, including an alt, and that calcullates the position of the transponder (fake) vehicle at specified distance and direction.

   But why do you need to modify alt_uav, the altitude of the vehicle?

Reply:
@hamishwillee You are correct. I was just thinking there are scenarios which maybe user wants to change the altitude of the current vehicle in simulation to play out with the fake traffic production. But it also seems confusing in most scenarios and might be an extra to do. 
I think it might be better to remove it for a better clarity?